### PR TITLE
1557 - Fixed the issue with group, members page loading with RTL lang…

### LINKED ISSUE #1557: You can not go to Forums setting while creating a group when using Persian forum slugs.
**Describe the bug**
When I want to create a group, the 1st and 2nd steps of the group creation form works fine, but from Forums step, it doesn't work. 

**To Reproduce**
Follow https://www.loom.com/share/1642b9a9643a4c249a878de5888a5fb4

**Expected behavior**

**Screenshots**
Provided 

**Support ticket links**
https://secure.helpscout.net/conversation/1285719008/99362?folderId=3701249

### PR DESCRIPTION
**Pull request by janakkaneriya007**
…uages

### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #1557.

### How to test the changes in this Pull Request:

1. Change groups, members page slug to Arabic, Persian OR any RTL languages
2. Go to groups, members page
3. click on URL bar and try to refresh by hitting enter key

### Proof Screenshots or Video
https://www.loom.com/share/d4d087e38850442995229598d500f072

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Review by dipskakadiya — DISMISSED** [review]
@janakkaneriya007 Approved. Please merge this PR to `dev` branch and assign to @mehulkaklotar for final review

File: src/bp-core/bp-core-catchuri.php [hunk 1]
```diff
@@ -84,6 +84,10 @@ function bp_core_set_uri_globals() {
 
 	// Take GET variables off the URL to avoid problems.
 	$path = strtok( $path, '?' );
+	
+	// Convert upper case to lower to avoid problem with assigning component
+	$path = strtolower( $path );
+
```
Comment (mehulkaklotar) [inline src/bp-core/bp-core-catchuri.php:90]: @janakkaneriya007 please fix the phpcs errors

```
L88: Squiz.Commenting.InlineComment.InvalidEndChar: Inline comments must end in full-stops, exclamation marks, or question marks

L90: Squiz.WhiteSpace.SuperfluousWhitespace.EmptyLines: Functions must not contain multiple empty lines in a row; found 2 empty lines

L91: Squiz.WhiteSpace.SuperfluousWhitespace.EndLine: Whitespace found at end of line
```

File: src/bp-forums/core/options.php [hunk 2]
```diff
@@ -486,6 +486,8 @@ function bbp_maybe_get_root_slug() {
 	if ( bbp_get_root_slug() && bbp_include_root_slug() )
 		$retval = trailingslashit( bbp_get_root_slug() );
 
+	$retval = urldecode_deep( $retval) ;
+
```
Comment (mehulkaklotar) [inline src/bp-forums/core/options.php:490]: @janakkaneriya007 

```
L486: Generic.ControlStructures.InlineControlStructure.NotAllowed: Inline control structures are not allowed
L489: PEAR.Functions.FunctionCallSignature.SpaceAfterCloseBracket: Space after closing parenthesis of function call prohibited
L489: PEAR.Functions.FunctionCallSignature.SpaceBeforeCloseBracket: Expected 1 spaces before closing parenthesis; 0 found
L489: Squiz.WhiteSpace.SemicolonSpacing.Incorrect: Space found before semicolon; expected ");" but found ") ;"
L490: Squiz.WhiteSpace.SuperfluousWhitespace.EndLine: Whitespace found at end of line
```

Comment (mehulkaklotar) [reply src/bp-forums/core/options.php:490]: @janakkaneriya007 

1. what if `$retval` is empty string?
2. what if function return value from filter `bbp_maybe_get_root_slug` is not urldecoded.

**Review by mehulkaklotar — CHANGES_REQUESTED** [review]

**Comment by janakkaneriya007** [comment]
Fixed here: https://github.com/buddyboss/buddyboss-platform/pull/1215

**Pull request closed without merging**